### PR TITLE
CPR-520 add syscon sync API definition

### PR DESCRIPTION
**Pull request by markreesmoj**

### CODE REVIEW

**Review by H-Iwanejko — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/model/sysconsync/Prisoner.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+data class Prisoner(
+  val title: String? = null,
+  val firstName: String? = null,
+  val middleName1: String? = null,
+  val middleName2: String? = null,
+  val lastName: String? = null,
+  val dateOfBirth: LocalDate? = null,
+  val birthplace: String? = null,
+  val nationality: String? = null,
+  val multipleNationalities: List<String> = emptyList(),
```
Comment (grnewbould) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/model/sysconsync/Prisoner.kt:14]: Should we just call this nationalities ?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/model/sysconsync/Prisoner.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  val crn: String? = null,
+  var prisonNumber: String? = null,
+  val ethnicity: String? = null,
+  val race: String? = null,
+  val sexualOrientation: String? = null,
+  val sex: String? = null,
+  val interestToImmigration: Boolean? = null,
+  val disability: Boolean? = null,
+  val currentlyManaged: Boolean? = null,
+  val sentenceStartDate: LocalDate? = null,
```
Comment (grnewbould) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/model/sysconsync/Prisoner.kt:25]: Do we have the ability to get multiple historic sentence dates, or, are we limited to 1 from Nomis ?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SysconSyncController.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import uk.gov.justice.digital.hmpps.personrecord.api.model.sysconsync.Prisoner
+import uk.gov.justice.hmpps.kotlin.common.ErrorResponse
+
+@Tag(name = "Syscon Sync")
+@RestController
+@PreAuthorize("hasRole('${Roles.QUEUE_ADMIN}')") // TODO change me :-)
+class SysconSyncController {
+
+  @Operation(description = "Create a prison record")
+  @PutMapping("/syscon-sync/{prisonNumber}")
```
Comment (grnewbould) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SysconSyncController.kt:29]: Also should have a post ?  and possibly a merge end-point ?

Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SysconSyncController.kt:29]: yes definitely 
Delete too
Model for post will be the same - the aim of this is just to get the model in place so Syscon can start work

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/model/sysconsync/Prisoner.kt (hunk 2 above)
Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/model/sysconsync/Prisoner.kt:25]: Good point, one to raise in the meeting tomorrow.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SysconSyncController.kt (hunk 3 above)
Comment (grnewbould) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SysconSyncController.kt:29]: Deletes are manual via a script I believe.  So not sure we will need to expose an API for that.

**Pull request merged**